lib/rubocop/cop/root_cops/factory_file_name.rb

```diff
@@ -0,0 +1,36 @@
+require "pry"
+
+module RuboCop
+  module Cop
+    module RootCops
+      class FactoryFileName < Cop
+        FILE_NAME_ERROR = "Factory name should match file name (one factory per file)".freeze
+        FACTORY = %i[factory].freeze
+
+        def investigate(processed_source)
+          file_path = processed_source.buffer.name
+          @base_name = File.basename(file_path, ".rb")
+        end
+
+        def on_send(node)
+          method_name = node.method_name
+          return unless _factory_method?(method_name)
+
+          # these 2 lines turn [:factory_name] into "factory_name"
+          factory_symbol_name = *node.arguments[0]
+          factory_name = factory_symbol_name[0].id2name
+
+          if factory_name != @base_name
+            add_offense(node, :location => :expression, :message => FILE_NAME_ERROR)
+          end
+        end
+
+        private
+
+        def _factory_method?(method)
+          FACTORY.include?(method)
```

Comment:
This is really minor, but why not `method == :factory` here instead of defining the array constant?

Reply:
Makes sense! I overcomplicated it a bit haha